lib/lib/network/response/error_response.dart

```diff
@@ -0,0 +1,11 @@
+class ErrorResponse implements Exception {
+  ErrorResponse({this.statusCode = -1, required this.reasonPhrase});
+
+  final int statusCode;
```

Comment:
Apróság csak de én nem -1 -nek égetném be a default értéket, hanem inkább `int?` típussal venném fel. Egyértelműbb lenne ha `null`-t  kapna alapból, mintha -1 lenne az alapérték